include/iota/api/extended.hpp

```diff
@@ -405,7 +405,9 @@ class Extended : public Core {
   Responses::SendTransfer promoteTransaction(const Types::Trytes& tail, int depth,
                                              int                            minWeightMagnitude,
                                              std::vector<Models::Transfer>& transfers, int delay,
-                                             const bool& interrupt) const;
+                                             const std::function<bool()>& interrupt = []() {
```

Comment:
I'm not sure about that default one.
I'd either not put any default function, or just put one that returns true and allow for maximum one iteration.
Overall, I don't see any benefit of making this parameter optional, this is definitely something the user has to implement based on his needs.